spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/StreamListenerDuplicateMappingTests.java

```diff
@@ -38,27 +39,73 @@
 
 	@Test
 	@SuppressWarnings("unchecked")
-	public void testDuplicateMapping() throws Exception {
+	public void testDuplicateMapping() {
+		ConfigurableApplicationContext context = null;
 		try {
-			ConfigurableApplicationContext context = SpringApplication.run(TestDuplicateMapping.class,
-					"--server.port=0");
+			context = SpringApplication.run(TestDuplicateMapping.class, "--server.port=0");
 			fail("Exception expected on duplicate mapping");
 		}
 		catch (BeanCreationException e) {
 			assertThat(e.getCause().getMessage()).startsWith("Duplicate @StreamListener mapping");
 		}
+		finally {
+			if (context != null) {
+				context.close();
+			}
+		}
+	}
+
+	@Test
+	public void testDuplicateMappingFromAbstractMethod() {
+		ConfigurableApplicationContext context = null;
+		try {
+			context = SpringApplication.run(TestDuplicateMappingFromAbstractMethod.class, "--server.port=0");
+		}
+		catch (BeanCreationException e) {
+			String errorMessage = e.getCause().getMessage().startsWith("Duplicate @StreamListener mapping") ?
+					"Duplicate mapping exception is not expected" : "Test failed with exception";
+			fail(errorMessage + ": " + e.getMessage());
+		}
+		finally {
+			if (context != null) {
+				context.close();
+			}
+		}
 	}
 
-	@EnableBinding(Processor.class)
+	@EnableBinding(Sink.class)
 	@EnableAutoConfiguration
 	public static class TestDuplicateMapping {
 
-		@StreamListener(Processor.INPUT)
+		@StreamListener(Sink.INPUT)
 		public void receive(Message<String> fooMessage) {
 		}
 
 		@StreamListener(Processor.INPUT)
 		public void receiveDuplicateMapping(Message<String> fooMessage) {
 		}
 	}
+
+	@EnableBinding(Sink.class)
+	@EnableAutoConfiguration
+	public static class TestDuplicateMappingFromAbstractMethod implements ISink<TestBase> {
+
+		@Override
+		@StreamListener(Sink.INPUT)
+		public void testMethod(TestBase msg) {
+		}
+	}
+
+	public interface ISink<T extends Base> {
```

Comment:
We can possibly use a different name for the interface (not a huge fan of `I`-prefixed interfaces :) ).